### PR TITLE
fix issue 1968 and correct visitor pattern code

### PR DESCRIPTION
**Pull request by xyllq999**
fix issue 1968 and correct visitor pattern code

Modified issue with visitor mode code, changed different method name to method reload, fixed issue1968.

Pull request description

The main modification of the method in the class is visit, which is accessed only by different parameter types, eliminating some errors and simplifying code writing, in line with the pattern requirements.

### CODE REVIEW

**Comment by xyllq999** [comment]
this PR helps for solve #1968

File: visitor/src/main/java/com/iluwatar/visitor/CommanderVisitor.java [hunk 1]
```diff
@@ -30,19 +30,32 @@
  */
 @Slf4j
 public class CommanderVisitor implements UnitVisitor {
+  //CS304 Issue link: https://github.com/iluwatar/java-design-patterns/issues/1968
```
Comment (ohbus) [inline visitor/src/main/java/com/iluwatar/visitor/CommanderVisitor.java:33]: Please remove this comment

File: visitor/src/main/java/com/iluwatar/visitor/Sergeant.java [hunk 2]
```diff
@@ -32,9 +32,14 @@ public Sergeant(Unit... children) {
...
   }
 
+  /**
+   * Accept a visitor.
+   * @param visitor An implementation class of Unit.
+   */
   @Override
   public void accept(UnitVisitor visitor) {
-    visitor.visitSergeant(this);
+    //CS304 Issue link: https://github.com/iluwatar/java-design-patterns/issues/1968
```
Comment (ohbus) [inline visitor/src/main/java/com/iluwatar/visitor/Sergeant.java:41]: Remove this comment

File: visitor/src/main/java/com/iluwatar/visitor/SergeantVisitor.java [hunk 3]
```diff
@@ -30,19 +30,32 @@
  */
 @Slf4j
 public class SergeantVisitor implements UnitVisitor {
+  //CS304 Issue link: https://github.com/iluwatar/java-design-patterns/issues/1968
```
Comment (ohbus) [inline visitor/src/main/java/com/iluwatar/visitor/SergeantVisitor.java:33]: Remove this comment

File: visitor/src/main/java/com/iluwatar/visitor/Soldier.java [hunk 4]
```diff
@@ -32,9 +32,14 @@ public Soldier(Unit... children) {
...
   }
 
+  /**
+   * Accept a visitor.
+   * @param visitor An implementation class of Unit.
+   */
   @Override
   public void accept(UnitVisitor visitor) {
-    visitor.visitSoldier(this);
+    //CS304 Issue link: https://github.com/iluwatar/java-design-patterns/issues/1968
```
Comment (ohbus) [inline visitor/src/main/java/com/iluwatar/visitor/Soldier.java:41]: Remove this comment

File: visitor/src/main/java/com/iluwatar/visitor/SoldierVisitor.java [hunk 5]
```diff
@@ -31,18 +31,32 @@
 @Slf4j
 public class SoldierVisitor implements UnitVisitor {
 
+  //CS304 Issue link: https://github.com/iluwatar/java-design-patterns/issues/1968
```
Comment (ohbus) [inline visitor/src/main/java/com/iluwatar/visitor/SoldierVisitor.java:34]: Remove this comment

File: visitor/src/main/java/com/iluwatar/visitor/UnitVisitor.java [hunk 6]
```diff
@@ -28,10 +28,24 @@
  */
 public interface UnitVisitor {
 
-  void visitSoldier(Soldier soldier);
+  //CS304 Issue link: https://github.com/iluwatar/java-design-patterns/issues/1968
```
Comment (ohbus) [inline visitor/src/main/java/com/iluwatar/visitor/UnitVisitor.java:31]: Remove this comment

File: visitor/src/test/java/com/iluwatar/visitor/CommanderTest.java [hunk 7]
```diff
@@ -42,7 +42,8 @@ public CommanderTest() {
 
   @Override
   void verifyVisit(Commander unit, UnitVisitor mockedVisitor) {
-    verify(mockedVisitor).visitCommander(eq(unit));
+    //CS304 (manually written) Issue link: https://github.com/iluwatar/java-design-patterns/issues/1968
```
Comment (ohbus) [inline visitor/src/test/java/com/iluwatar/visitor/CommanderTest.java:45]: Remove this comment

File: visitor/src/test/java/com/iluwatar/visitor/SoldierTest.java [hunk 8]
```diff
@@ -42,7 +42,8 @@ public SoldierTest() {
 
   @Override
   void verifyVisit(Soldier unit, UnitVisitor mockedVisitor) {
-    verify(mockedVisitor).visitSoldier(eq(unit));
+    //CS304 (manually written) Issue link: https://github.com/iluwatar/java-design-patterns/issues/1968
```
Comment (ohbus) [inline visitor/src/test/java/com/iluwatar/visitor/SoldierTest.java:45]: Remove this comment

**Review by ohbus — CHANGES_REQUESTED** [review]
Thank you so much for your contributions!!

Please address the review comments and possibly eliminate the sonar issues as reported in the comment and make it a spotless PR.

Looking forward to this!

File: visitor/src/test/java/com/iluwatar/visitor/SoldierTest.java (hunk 8 above)
Comment (xyllq999) [reply visitor/src/test/java/com/iluwatar/visitor/SoldierTest.java:45]: Thanks a lot for mention that, I have removed these comment, sorry about it.

**Comment by xyllq999** [comment]
@ohbus I have remove all the comment, thanks a lot!

**Pull request closed without merging**